softwareconfiguration/model/src/main/java/ai/libs/softwareconfiguration/model/ComponentInstance.java

```diff
@@ -1,329 +1,327 @@
+	/**
+	 * This method checks, whether a path of refinements conforms the constraints for parameter refinements.
+	 *
+	 * @param path
+	 *            A path of refinements to be checked.
+	 * @return Returns true if everything is alright and false if there is an issue with the given path.
+	 */
+	public boolean matchesPathRestriction(final List<Pair<String, String>> path) {
+		if (path.isEmpty()) {
+			return true;
+		}
+
+		/* if the first entry is on null, we interpret it as a filter on this component itself */
+		int i = 0;
+		if (path.get(0).getX() == null) {
+			String requiredComponent = path.get(0).getY();
+			if (!requiredComponent.equals("*") && !this.component.getName().equals(requiredComponent)) {
+				return false;
+			}
+			i = 1;
+		}
+
+		/* now go over the rest of the path and check every entry on conformity */
+		ComponentInstance current = this;
+		int n = path.size();
+		for (; i < n; i++) {
+			Pair<String, String> selection = path.get(i);
+			if (!current.getComponent().getRequiredInterfaces().containsKey(selection.getX())) {
+				throw new IllegalArgumentException("Invalid path restriction " + path + ": " + selection.getX() + " is not a required interface of " + current.getComponent().getName());
+			}
+			ComponentInstance instanceChosenForRequiredInterface = current.getSatisfactionOfRequiredInterfaces().get(selection.getX());
+			if (!selection.getY().equals("*") && !instanceChosenForRequiredInterface.getComponent().getName().equals(selection.getY())) {
+				return false;
+			}
+			current = instanceChosenForRequiredInterface;
+		}
+		return true;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((this.component == null) ? 0 : this.component.hashCode());
+		result = prime * result + ((this.parameterValues == null) ? 0 : this.parameterValues.hashCode());
+		result = prime * result + ((this.satisfactionOfRequiredInterfaces == null) ? 0 : this.satisfactionOfRequiredInterfaces.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(final Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null) {
+			return false;
+		}
+		if (this.getClass() != obj.getClass()) {
+			return false;
+		}
+		ComponentInstance other = (ComponentInstance) obj;
+		if (this.component == null) {
+			if (other.component != null) {
+				return false;
+			}
+		} else if (!this.component.equals(other.component)) {
+			return false;
+		}
+		if (this.parameterValues == null) {
+			if (other.parameterValues != null) {
+				return false;
+			}
+		} else if (!this.parameterValues.equals(other.parameterValues)) {
+			return false;
+		}
+		if (this.satisfactionOfRequiredInterfaces == null) {
+			if (other.satisfactionOfRequiredInterfaces != null) {
+				return false;
+			}
+		} else if (!this.satisfactionOfRequiredInterfaces.equals(other.satisfactionOfRequiredInterfaces)) {
+			return false;
+		}
+		return true;
+	}
+
+	public String toComponentNameString() {
```

Comment:
Do we really want to have this method in a ComponentInstance or does this rather belong to a util class?